internal/state/secrets/secrets.go

```diff
@@ -74,6 +80,8 @@ type SecretDiskMemoryManager interface {
 	Request(nsname types.NamespacedName) (string, error)
 	// WriteAllRequestedSecrets writes all requested secrets to disk.
 	WriteAllRequestedSecrets() error
+	// GetAllRequestedSecrets returns all request secrets as Files.
+	GetAllRequestedSecrets() []File
```

Comment:
@pleshakov 

I left the `WriteAllRequestedSecrets` method because I'm toying with the idea of implementing the `ConfigStorer`, `Subject`, and `Observer` interfaces for configuring an nginx instance in the same Pod. That would allow us to benchmark the two approaches. What do you think?

Either way, I still need to rename this interface, or separate the Get functionality from the Write.

Reply:
Not sure yet... it would probably be an e2e test where we benchmark the time it takes to update the config in the dataplane.